core-web/libs/sdk/react/src/lib/hooks/useDotcmsEditorPageAsset.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react';
+
+import { onFetchPageAssetFromUVE } from '@dotcms/client';
+
+import { DotCMSPageContext } from '../models';
+
+export const useDotcmsEditorPageAsset = (
+    currentPageAsset = {} as DotCMSPageContext['pageAsset']
+) => {
+    const [pageAsset, setPageAsset] = useState<DotCMSPageContext['pageAsset'] | null>(null);
+
+    useEffect(() => {
+        const cleanListener = onFetchPageAssetFromUVE((page) => {
+            setPageAsset(page as DotCMSPageContext['pageAsset']);
```

Comment:
Do we need to care if `page` is null or undefined?

Reply:
Probably no, because if the page is null, i think the UVE redirect to the browser page
But just in case i added the check if page is different to null | undefined